gorapass/views.py

```diff
@@ -23,9 +35,49 @@ def hike(request, hike_id):
 
 def hikes(request):
     hike_models = Hikes.objects.all()
+
+    # Filter out hikes if there is filtration criteria in the request
+    if request.body:
+        filters = json.loads(request.body)['filters']
+        if not valid_hike_filters(filters):
+            return HttpResponseBadRequest('Invalid filtration criteria')
+
+        hike_models = filter_hikes(hike_models, filters)
+
     hike_dicts = [ model_to_dict(hike) for hike in hike_models ]
     return JsonResponse(hike_dicts, safe=False)
 
+def valid_hike_filters(filters):
+    for filter in filters:
+        if set(filter.keys()) != FILTER_KEYS:
+            return False
+        if not hasattr(Hikes, filter[ATTRIBUTE_NAME]):
```

Comment:
There's a chance in the future that the returned filters payload will be a JSON with all the different filters listed and just a null value next to the filter that isn't used. I think this is fine for now but we might want to revisit this in the future depending on how we end up setting up the post request for filters

Reply:
Yeah could be. Things also change if we end up with hikes that have different attributes.